lib/grammers-mtsender/src/lib.rs

```diff
@@ -68,7 +76,7 @@ pub(crate) fn generate_random_id() -> i64 {
 
 // Manages enqueuing requests, matching them to their response, and IO.
 pub struct Sender<T: Transport, M: Mtp> {
-    stream: TcpStream,
+    stream: Box<dyn Stream>,
```

Comment:
This is an unfortunate (tiny) performance hit for the use-case where no proxies are used, but that can be solved with a new generic parameter.

Reply:
Do you have any suggestions here?

Reply:
No, it's fine to leave this as-is for now.